File: Java/Elektronik.java
// Elektronik.java
public abstract class Elektronik {
    protected String merk;
    protected String nama;
    protected double harga;
    
    public Elektronik() {
        this.merk = "";
        this.nama = "";
        this.harga = 0.0;
    }
    
    public Elektronik(String merk, String nama, double harga) {
        this.merk = merk;
        this.nama = nama;
        this.harga = harga;
    }
    
    public void setMerk(String merk) {
        this.merk = merk;
    }
    
    public void setNama(String nama) {
        this.nama = nama;
    }
    
    public void setHarga(double harga) {
        this.harga = harga;
    }
    
    public String getMerk() {
        return merk;
    }
    
    public String getNama() {
        return nama;
    }
    
    public double getHarga() {
        return harga;
    }
    
    public abstract void tampilkanInfo();
}

File: CPP/Elektronik.cpp
#pragma once
#include <iostream>
#include <string>

using namespace std;

// Base class for all electronic components
class Elektronik {
protected:
    string merk;
    string nama;
    float harga;

public:
    Elektronik() {}

    Elektronik(string merk, string nama, float harga) {
        this->merk = merk;
        this->nama = nama;
        this->harga = harga;
    }

    void setMerk(string merk) {
        this->merk = merk;
    }

    void setNama(string nama) {
        this->nama = nama;
    }

    void setHarga(float harga) {
        this->harga = harga;
    }

    string getMerk() const {
        return merk;
    }

    string getNama() const {
        return nama;
    }

    float getHarga() const {
        return harga;
    }

    virtual void tampilkanInfo() {
        cout << "Elektronik: " << merk << " " << nama << ", Harga: Rp" << harga << endl;
    }

    virtual ~Elektronik() {}
};